db.js

```diff
@@ -0,0 +1,519 @@
+  ARRAY,
+  STRING,
+  DECIMAL,
+  INTEGER,
+  UUID,
+  UUIDV4,
+  DATE,
+  ENUM
+} = Sequelize;
+
+const conn = new Sequelize(
+  process.env.DATABASE_URL || 'postgres://localhost/teamb_graceshopperdb'
+);
+
+//USERS COMMENTED OUT UNTIL PRODUCTS WORKS
+
+const User = conn.define('user', {
+  id: {
+    type: UUID,
+    primaryKey: true,
+    defaultValue: UUIDV4
+  },
+  username: {
+    type: STRING,
+    notEmpty: true,
+    notNull: true,
+    unique: true
+  },
+  password: {
+    type: STRING,
+    notNull: true,
+    notEmpty: true
+  },
+  email: {
+    type: STRING,
+    notEmpty: true,
+    notNull: true,
+    unique: true,
+    validate: {
+      isEmail: true
+    }
+  },
+  firstName: {
+    type: STRING
+  },
+  lastName: {
+    type: STRING
+  },
+  streetAddress: {
+    type: STRING
+  },
+  city: {
+    type: STRING
+  },
+  state: {
+    type: STRING
+  },
+  zipcode: {
+    type: INTEGER
+  },
+  billStreetAddress: {
+    type: STRING
+  },
+  billCity: {
+    type: STRING
+  },
+  billState: {
+    type: STRING
+  },
+  billZipcode: {
+    type: INTEGER
+  }
+});
+
+const Guest = conn.define('guest', {
+  id: {
+    type: UUID,
+    primaryKey: true,
+    defaultValue: UUIDV4
+  },
+  cart: {
+    type: ARRAY(INTEGER),
+    defaultValue: []
+  },
+  wishlist: {
+    type: ARRAY(INTEGER),
+    defaultValue: []
+  }
+});
+
+const Product = conn.define('product', {
+  id: {
+    type: UUID,
+    primaryKey: true,
+    defaultValue: UUIDV4
+  },
+  productName: {
+    type: STRING,
+    notNull: true,
+    notEmpty: true
+  },
+  description: {
+    type: TEXT,
+    defaultValue: 'Please contact us for more details...'
+  },
+  price: {
+    type: DECIMAL,
+    notEmpty: true,
+    defaultValue: 99.99
+  },
+  imageURL: {
+    //STRING FOR NOW BUT COULD BE AN ARRAY IF MULTIPLE PHOTOS
+    type: STRING,
+    defaultValue: 'https://live.staticflickr.com/15/89773667_f91cdd7c11.jpg'
+  },
+  inventory: {
+    type: INTEGER,
+    defaultValue: 1,
+    validate: {
+      min: 0
+    }
+  }
+});
+
+//==============================ORDERS?==============================
+//==============HOW TO HANDLE ORDERS? SEPERATE MODEL OR ARRAY ON USER?
+
+const Order = conn.define('order', {
+  id: {
+    type: UUID,
+    primaryKey: true,
+    defaultValue: UUIDV4
+  },
+  status: {
+    type: ENUM('cart', 'completed'),
+    defaultValue: 'cart'
+  },
+  orderDate: {
+    type: DATE
+  },
+  shippingAddress: {
+    type: TEXT
+  },
+  total: {
+    type: DECIMAL,
+    defaultValue: 0.0
+  }
+});
+
+//==============================CART?==============================
+//==============HOW TO HANDLE CART? SEPERATE MODEL OR ARRAY ON USER?
+//==============SHOULD CART BY HANDLED BY STORE AND FRONT END? WILL THIS BE PERSISTENT?
+const OrderProducts = conn.define('orderproducts', {
+  id: {
+    type: UUID,
+    primaryKey: true,
+    defaultValue: UUIDV4
+  },
+  quantity: {
+    type: INTEGER,
+    validate: {
+      min: 1
+    }
+  },
+  price: {
```

Comment:
I don't think that we need price anywhere but on the product, and maybe a record of a purchase - which is kind of your `Order`.

Price is just Product Price * Quantity. Which you can create a virtual field on this for, so that it appears here.